test/oracle.test.ts

```diff
@@ -15,9 +16,57 @@ contract('Oracle', ([owner]) => {
     oracle = await Oracle.new({from: owner})
   })
 
-  describe('oracle', () => {
-    it('oracle', async () => {
-      //empty
+  describe('Batch oracle', () => {
+    it('should revert setting batch oracle from non-owner address', async () => {
+      await expectRevert(oracle.setBatchOracle(batch, batchOracle, {from: random}), 'Ownable: caller is not the owner')
+    })
+
+    it('should set batch oracle', async () => {
+      await oracle.setBatchOracle(batch, batchOracle, {from: owner})
+
+      assert.equal(await oracle.getBatchOracle(batch), batchOracle, 'batch oracle address mismatch')
+    })
+  })
+
+  describe('Oracle locking period', () => {
+    const lockingPeriod = new BigNumber(60 * 15) // 15min
+
+    it('should revert setting oracle locking period from non-owner address', async () => {
+      await expectRevert(
+        oracle.setLockingPeriod(batchOracle, lockingPeriod, {from: random}),
+        'Ownable: caller is not the owner',
+      )
+    })
+
+    it('should set oracle locking period', async () => {
```

Comment:
Is there a max time locking period that can be set? We might need some restrictions. Also is the owner manually setting locking period?!

Reply:
I don't think we discussed a max time locking period ? if so what is the limit that I should test ?
and re owner, yeah the owner manually setting locking period.

Reply:
maybe 3 days max, 15 minutes min?

Reply:
hmm so how is the locking / dispute period set before the prices are set or how does the overall flow work?